src/metabase/cloud_migration/models/cloud_migration.clj

```diff
@@ -226,6 +230,12 @@
       (log/info "Notifying store that upload is done")
       (http/put (migration-url external_id "/uploaded"))
 
+      ;; Need to restore the previous scheduler configuration because the database quartz is pointing at has changed
+      ;; after finishing the dump to h2 migration
+      (task.bootstrap/set-jdbc-backend-properties! (mdb/db-type))
```

Comment:
The jdbc adapter for quartz is controlled through a system property and it changed when the 'temp' scheduler macro runs without being reset afterwards.